### DISCUSSION TITLE
BrickRail on Mac and Lego Remote

### DISCUSSION THREAD

**Discussion by TekyTeky** [Novakasa/brickrail#115] [General]
I just saw your discussion on Eurobricks.

I have a Mac so can do any testing if you want.. I only got the PC to run BAP.
Although I work in IT, application support, so use Windows at work too.
Mac is my preferred at home OS.

Regarding using a Lego PU remote, the guy on that post Lok24, made a Pybricks program that I use on one of my trains. It uses the Lego remote with no change to firmware on it. It's good because it can run very slowly, for shunting or faster for normal running. I can provide you a copy as I am not sure if he still updates it.

**Comment by Novakasa** [comment]
Oh, good to know! I'll see what can be done about a MAC port, I unfortunately have never really used one, so building a executable for it requires a bit of research on my part. But it would definitely be helpful to have someone who can test my "experiments"!

About the remote thing, yeah I know of that script and I have no problem implementing the remote logic itself, the problem is rather that the RAM of a City hub is actually very constrained. With the brickrail program running on the hub, I'm already really close to the limit and I need to be very careful with what things I add to the hub programs.

For now, I think the easiest would be to implement a better system to control the trains in the brickrail GUI itself.
Also, I could imagine a path where a remote is just connected to one of the layout hubs, which talks to the PC, which then controls the train.

In any way all these things are unfortunately not coming too soon, as I definitely have a lot of different areas that need more polish.

**Reply by Novakasa** [reply]
Of course another possibility would be to just run a different program when the user wants to manually control the train, but that does open up a lot of questions about how to really incorporate it into the gui, but certainly not impossible to figure out

**Reply by TekyTeky** [reply]
Well I have 2 Windows laptops, so I can still run BAP and also run Brickrail. 
Then I can have a random couple of trains doing the show stuff using Brickrail and BAP for my shunting. So this makes me very happy. 😎

**Comment by Elphiel01** [comment]
Hi,
Just registered here to follow your software and looking forward to a mac release too! Your easy Interface really appeals to me as mac user and is the first automation software that even a non programmer like me wants to use. In the meantime I start sourcing some cheap china WeDo2.0 motors for my switches and maybe find someone who could lend me an old windows laptop to give it already a try.

**Reply by Novakasa** [reply]
Thanks for following the project!

I already did some attempts and had my brother try to start the project on his macbook. Currently, the GUI can be run, but the Bluetooth server seems to be more tricky to get running. Main issue is the apple notarization and code signing process, Macbooks don't seem to want to run non-signed code and adding exceptions for a non .app bundle program seems to be a struggle. I hopefully can investigate more when I see my brother in a few weeks and can use his macbook for a bit.
